Debug/Assertion.hpp

```diff
@@ -200,11 +200,9 @@ template <typename T>
 
 #define DEBUG_CODE(X) {}
 
-#define __IGNORE_WUNUSED(...) __PUSH_DIAGNOSTICS("GCC diagnostic ignored \"-Wreturn-type\"", __VA_ARGS__)
-
 #define ASS(Cond)  {}
-#define ALWAYS(Cond) (void) ( Cond );
-#define NEVER(Cond) (void) ( Cond );
+#define ALWAYS(Cond) if (!(Cond)) __UNREACHABLE;
+#define NEVER(Cond) if ((Cond)) __UNREACHABLE;
```

Comment:
I assert (without evidence but with reasonable confidence) this has a minimal, if not positive, impact on code generation. It persuades GCC that quite a few warnings aren't problems because they aren't reachable.